File: tests/06_phases_openmp.dg/nanox/cxx/success_task_02.cpp
/*
<testinfo>
test_generator=config/mercurium-nanox
</testinfo>
*/

template <typename T>
struct A
{
    T e;
};

struct B
{
    int d;
};

void f(void)
{
    A<B> c;
#pragma omp task
    {
        c.e.d = 3;
    }
}

File: tests/07_phases_omp.dg/common/fortran/success_task_02.f90
! <testinfo>
! test_generator=config/mercurium-omp
! </testinfo>
MODULE M2
IMPLICIT NONE
  INTEGER :: X
END MODULE M2

PROGRAM P
USE M2
IMPLICIT NONE

X = 42
!$OMP TASK
    X = 1
!$OMP END TASK

!$OMP TASKWAIT

IF (X /= 1) STOP 1
END PROGRAM P
